### ISSUE TITLE
Access to possible undefined value not flagged as error

### ISSUE THREAD

**Issue by bpasero** [microsoft/TypeScript#37979]
**TypeScript Version:**  3.8.3

**Search Terms:**
strict null

**Code**

```ts
class Foo {
    
    private bar: Bar;

    readonly barProp = this.bar.prop;

    constructor() {
        this.bar = new Bar();
    }
}

class Bar {
    readonly prop = false;
}

const foo = new Foo();
console.log(foo.barProp);
```

**Expected behavior:**
A compile error is reported for accessing `this.bar.prop;`

**Actual behavior:**
No compile error reported.

**Playground Link:** https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=18&pc=26#code/MYGwhgzhAEBiD29oG8BQ0PXZgDgJwEsA3MAFwFNoAjMPALmgCFaBuVbDPcsAE3gDsQAT2q0ACnng5oAXmikAFgQgA6GnhX4pbDtGACIpPAFdgpeHgAUAShS7M8pavWzo-cgHcmtG2wcBfVEDUUEgYZjw7By5eAWFoLWk5ADMwEAhyNmCQg1JoZMRXdy8EeF8c-gh4EHIVEHgAc0sC+DVxSRxrFiA

**Comment by ahejlsberg** [comment]
I agree we should error here. Specifically, a `this.xxx` property access in a property initializer that references a property previously previously in the same class should be an error if that property has no initializer and no `!` definite assignment assertion.
